rundeckapp/grails-app/services/rundeck/services/ExecutionService.groovy

```diff
@@ -1041,11 +1041,24 @@ class ExecutionService implements ApplicationContextAware, StepExecutor, NodeSte
 //        }
 //
 
-        //set up log output threshold
-        def thresholdMap = ScheduledExecution.parseLogOutputThreshold(scheduledExecution?.logOutputThreshold)
-        def threshold = LoggingThreshold.fromMap(thresholdMap,scheduledExecution?.logOutputThresholdAction)
-
-        def ExecutionLogWriter loghandler= loggingService.openLogWriter(
+        // Output limit
+        def globalThresholdMap = getGlobalThresholdLimit()
+        def jobThresholdMap    = getJobThresholdLimit(scheduledExecution)
+        def thresholdMap       = LoggingThreshold.getOutputLimit(globalThresholdMap, jobThresholdMap)
+
+        // Output limit action
+        String limitAction = LoggingThreshold.getLimitAction(
+                globalThresholdMap,
+                getGlobalLimitAction(),
+                jobThresholdMap,
+                getJobLimitAction(scheduledExecution))
+        LoggingThreshold threshold = LoggingThreshold.fromMap(thresholdMap, limitAction)
+
+        String warnSize = configurationService.getString(LoggingThreshold.EXECUTION_LOGS_SIZE_WARNING, LoggingThreshold.WARN_SIZE_DEFAULT)
+        Map<String, Long> warnSizeMap = ScheduledExecution.parseLogOutputThreshold(warnSize)
+        threshold?.warningSize = warnSizeMap[LoggingThreshold.MAX_SIZE_BYTES] ?: warnSizeMap[LoggingThreshold.MAX_LINES]
```

Comment:
it seems like you are trying to define a secondary threshold for warning, using the same format as the normal threshold.  However, then you are setting the size value based on that format, when the threshold unit value may be different.  

Was the warning part of the customer request? It doesn't seem very useful, I think it should be removed

Reply:
I have the answer to that, the warning came from the customer and CS as a way to monitor this situation using Splunk or any monitor tool. Not mandatory, just valuable for the customer.